contracts/optimistic-ethereum/libraries/utils/Lib_BytesUtils.sol

```diff
@@ -170,6 +170,20 @@ library Lib_BytesUtils {
         return slice(_bytes, _start, _bytes.length - _start);
     }
 
+    function toBytes32PadLeft(
+        bytes memory _bytes
+    )
+        internal
+        pure
+        returns (bytes32)
+    {
+        bytes32 ret;
+        assembly {
+            ret := shr(mul(sub(32, mload(_bytes)), 8), mload(add(_bytes, 32)))
```

Comment:
I am worried this will underflow if `_bytes.length > 32`.  Not sure how we want to handle this or what expected behavior should be if that condition is true--probably just truncate?

Reply:
Ah we should truncate. Will update.